spyder/widgets/findreplace.py

```diff
@@ -329,7 +345,8 @@ def find_previous(self):
     def text_has_been_edited(self, text):
         """Find text has been edited (this slot won't be triggered when 
         setting the search pattern combo box text programmatically"""
-        self.find(changed=True, forward=True, start_highlight_timer=True)
+        if not self.replace_widgets[0].isVisible() or len(to_text_string(self.editor.get_selected_text()).splitlines())<=1:

```

Comment:
Add a comment for explaining this behavior too: "When selecting several lines, and replace box is activated, dynamically search is deactivated, for avoid changing the selection."